### DISCUSSION TITLE
How to suppress the omnipresent "Cancel" option in modal showInformationMessage

### DISCUSSION THREAD

**Discussion by msorens** [microsoft/vscode-discussions#457] [Extension Development QnA] [answered]
I was very surprised the first time I used the `modal` option for `showInformationMessage` and found it automatically adds "Cancel" as the final choice. How can I suppress that "Cancel" choice?

```
    const selection = await IDE.showInformationMessageModal(
      'Do you already have an account?', 'Yes', 'No');
 ```

<img width="337" alt="image" src="https://user-images.githubusercontent.com/6817500/218574275-a52e9013-f2df-4b75-acbf-f9b13b61ec72.png">

**Comment by gjsjohnmurray** [comment, marked as answer]
You need to set `isCloseAffordance: true` on the Message Item that is the button you want clicked when the user cancels the dialog (e.g. by pressing Esc)

See https://code.visualstudio.com/api/references/vscode-api#MessageItem

**Comment by wy-luke** [comment]
> You need to set `isCloseAffordance: true` on the Message Item that is the button you want clicked when the user cancels the dialog (e.g. by pressing Esc)
> 
> See [code.visualstudio.com/api/references/vscode-api#MessageItem](https://code.visualstudio.com/api/references/vscode-api#MessageItem)

Detailed version:

```typescript
const buttonOptions = [
    { title: 'Always' },
    { title: 'Never' },
    { title: 'NewCancle', isCloseAffordance: true },
]
const messageOptions: vscode.MessageOptions = { modal: true }

const pick = (await vscode.window.showWarningMessage(
    'Please choose',
    messageOptions,
    ...buttonOptions,
)) || { title: 'Cancle' }
```
